### ISSUE TITLE
Swift throwing methods produce wrong screenshot names

### ISSUE THREAD

**Issue by garnett** [uber/ios-snapshot-test-case#18]
In some of the tests I use this approach to avoid force-casting and alike: http://ericasadun.com/2017/05/27/tests-that-dont-crash/

Before it, my test looked like:

```swift
func test_layout() {
 // some setup, involving ! force unwrapping
 FBSnapshotVerifyView(view)
}
```
which produces screenshot with name based on method signature `test_layout_iPhone_375x812_en@3x.png` 

Now, I want to avoid using `!`:

```swift
func test_layout() throws {
  // code with safe unwrapping
  FBSnapshotVerifyView(view)
}
```

test still compiles and executes just fine but now screenshot name is `test_layoutAndReturnError:_iPhone_375x812_en@3x.png` due to objc error handling bridge, which causes test to fail.

I think it is safe to assume that `AndReturnError:` (with `:` standind for parameter, that's imprtant) could be stripped by `FBSnapshotTestCase` as I see `throws` pattern to be quite ubiquitous among iOS devs.

Happy to work on a PR if proposed solution makes sense.

**Comment by alanzeino** [comment]
Yes I think this can be solved in the library as it hurts Swift compatibility. Happy to take a PR.
